src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -280,6 +280,35 @@ public Response getDataset(@PathParam("datasetId") Integer datasetId){
         }
     }
 
+    @GET
+    @Path("/batch")
+    @Produces("application/json")
+    @PermitAll
+    public Response getDatasets(@QueryParam("ids") List<Integer> datasetIds){
+        try {
+            List<Dataset> datasets = datasetService.getDatasets(datasetIds);
+            if (datasets.size() != datasetIds.size()) {
```

Comment:
Size is one way to do this, but can be an error in the case where a user provides the same id more than once (i.e. ` 1, 1, 2, 3` -> list of datasets will be 3). 

One thing to do would be to ensure distinctness on the provided list.

Another option might look something like this:
```
List<Integer> foundIds = datasets.stream().map(Dataset::getDataSetId).toList();
if (!foundIds.containsAll(datasetIds)) { 
  // find the differences
  List<Integer> differences = new ArrayList<>(datasetIds);
  differences.removeAll(foundIds);
  ... throw error with differences ... 
}
```

Reply:
Testing with a duplicate ID that is valid shows this response:
```
{
  "message": "Could not find datasets with ids: ",
  "code": 404
}
```